### ISSUE TITLE
Error was not caught ReferenceError: require is not defined with 7.0

### ISSUE THREAD

**Issue by joaogarin** [zeit/next.js#5238]
# Examples bug report

## Example name
With apollo

## Describe the bug
VM64:37 Error was not caught ReferenceError: require is not defined
    at Module../node_modules/graphql/jsutils/instanceOf.mjs (instanceOf.mjs:31)

## To Reproduce
Steps to reproduce the behavior, please provide code snippets or a repository:
1. Go to https://github.com/zeit/next.js/tree/canary/examples/with-apollo
2.  clone and npm install
3. Launch site on http://localhost:3000

I can imagine other apollo examples also have the same issue after 7.x

## Expected behavior
No errors.

## Screenshots
<img width="1436" alt="screen shot 2018-09-20 at 16 57 34" src="https://user-images.githubusercontent.com/2866604/45831051-4f1bd180-bcf6-11e8-97a2-30374eafaf31.png">

## System information
 - OS: [macOS]
 - Browser (if applies) [e.g. chrome, safari]
 - Version of Next.js: [e.g. 7.0.0]

## Additional context

Only happens when trying out 7.0. It seems related to webpack4 and the support on .mjs files..it could be also an issue with graphql-js, I cant really find anything there. maybe someone else can help finding this out..will gladly move issue there if thats the case. Also dont know if at this point 7.0 is stable..just wanted to see how bad an upgrade would be, probably still a bit soon for me ;)

**Comment by amytych** [comment]
I solved it by updating `graphql`, it's quite outdated in this example. I sent a PR https://github.com/zeit/next.js/pull/5242 to fix it.

**Comment by joaogarin** [comment]
awesome, yeah I though graphql dependency was coming from react-apollo, but y that totally works tks 👍 cheers

**Comment by alexindigo** [comment]
We had exactly same issue, and upgrading graphql version didn't help (at first), so we spent quite some time investigating, until we've found that one of the dependency of our dependency had it's own copy of `graphql` that was older version – that package had graphql as `dependency` instead of listing it as `peerDependency`. After it was fixed everything went back to normal.

In case anybody else would be struggling with the same issue.
